container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go

```diff
@@ -1121,7 +1121,46 @@ func (manager *DockerManager) GetContainersByLabels(ctx context.Context, labels
 	return result, nil
 }
 
+// [FetchImage] always attempts to retrieve the latest [dockerImage].
+// If retrieving the latest [dockerImage] fails, the local image will be used.
+// Returns error, if no local image is available after retrieving latest fails.
 func (manager *DockerManager) FetchImage(ctx context.Context, dockerImage string) error {
```

Comment:
@tedim52 is there any way to return the result of fetching the image? It seems a little strange to call this function but not knowing if it was fetched successfully or not (other than the absence of the error)? Maybe you are limited by what the `isImageAvailableLocally` and `pullImage` functions return, so maybe there's not much you can do, but maybe worth having a look at?

Reply:
great point, send some ideas over slack -> lmk what you think and i'm happy to make this change. gonna merge for now and open another PR for this